src/transformers/models/jetmoe/configuration_jetmoe.py

```diff
@@ -0,0 +1,140 @@
+        num_key_value_heads (`int`, *optional*, defaults to 16):
+            Number of attention heads for each key and value in the Transformer encoder.
+        kv_channels (`int`, *optional*, defaults to 128): Defines the number of channels for the key and value tensors.
+        ffn_hidden_size (`int`, *optional*, defaults to 5632): Defines the hidden size of the feed-forward layer.
+        max_position_embeddings (`int`, *optional*, defaults to 4096):
+            The maximum sequence length that this model might ever be used with. JetMoE's sliding window attention
+            allows sequence of up to 4096*32 tokens.
+        activation_function (`string`, *optional*, defaults to `"silu"`): Defines the activation function for MLP experts.
+        glu (`bool`, *optional*, defaults to `True`): Whether to use Gated Linear Units in the MLP experts.
+        moe_num_experts (`int`, *optional*, defaults to 8): Defines the number of experts in the mixture of experts.
+        moe_top_k (`int, *optional*, defaults to 2): Defines the number of experts to use for each token.
+        use_cache (`bool`, *optional*, defaults to `True`):
+            Whether or not the model should return the last key/values attentions (not used by all models). Only
+            relevant if `config.is_decoder=True`.
+        bos_token_id (`int`, *optional*, defaults to 1):
+            The id of the "beginning-of-sequence" token.
+        eos_token_id (`int`, *optional*, defaults to 2):
+            The id of the "end-of-sequence" token.
+        tie_word_embeddings (`bool`, *optional*, defaults to `True`):
+            Whether the model's input and output word embeddings should be tied.
+        bias (`bool`, *optional*, defaults to `True`): Whether to use bias in the feed-forward and attention layer.
+        rope_theta (`float`, *optional*, defaults to 10000.0):
+            The base period of the RoPE embeddings.
+        rms_norm_eps (`float`, *optional*, defaults to 1e-06):
+            The epsilon used by the rms normalization layers.
+        initializer_range (`float`, *optional*, defaults to 0.01):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+
+    ```python
+    >>> from transformers import JetMoEModel, JetMoEConfig
+
+    >>> # Initializing a JetMoE 4B style configuration
+    >>> configuration = JetMoEConfig()
+
+    >>> # Initializing a model from the JetMoE 4B style configuration
+    >>> model = JetMoEModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    model_type = "jetmoe"
+    keys_to_ignore_at_inference = ["past_key_values"]
+
+    def __init__(
+        self,
+        vocab_size=32000,
+        hidden_size=2048,
+        num_hidden_layers=12,
+        num_attention_heads=32,
+        num_key_value_heads=16,
+        kv_channels=128,
```

Comment:
Thanks for the suggestion! The JetMoe architecture is a bit different. It uses a Mixture of Attention heads (MoA) layer that activates 2 group of query and output projections per layer. So similar to a MoE with top2 activation, MoA doubles the number of heads inside each attention layer, that's why we use a seperate kv_channels variable to avoid confusion.